File: PrepareAdaptiveFile.R
PrepareAdaptiveFile <- function(indir,outdir,thr=10000){
  ffs=dir(indir,full.names=T)
  for(ff in ffs){
    if(length(grep('\\.tsv',ff))==0)next
    ff0=unlist(strsplit(ff,'\\/'))
    ff0=ff0[length(ff0)]
    if(!file.exists(outdir))dir.create(outdir)
    newff=paste(outdir,'TestReal-',ff0,sep='')
    if(exists(newff))next
    ddnew=read.table(ff,header=T,sep='\t',stringsAsFactors=F)
    tmp.vv.rm=grep('\\*',ddnew[,2])
    if(length(tmp.vv.rm)>0)ddnew=ddnew[-tmp.vv.rm,]
    tmp.vv.rm=grep('X',ddnew[,2])
    if(length(tmp.vv.rm)>0)ddnew=ddnew[-tmp.vv.rm,]
    tmp.nn=nchar(ddnew[,2])
    tmp.vv=which(tmp.nn>=10 & tmp.nn<=24)
    #tmp.vv=which(tmp.nn>=1)
    ddnew=ddnew[tmp.vv,]
    ddnew=ddnew[which(ddnew[,6]!='unresolved'),]
    ddnew=ddnew[grep('^C.+F$',ddnew[,2]),]
    #ss=sum(ddnew[,3])
    #ddnew[,4]=ddnew[,3]/ss
    ddnew=ddnew[order(ddnew[,4],decreasing=T),]
    #q75=quantile(ddnew[,3],0.75)
    #THR=min(ddnew[,3]+1)
    #tmp.vv=which(ddnew[,3]>max(q75,THR))
    #if(length(tmp.vv)<=thr)tmp.vv=1:thr
    if(nrow(ddnew)>thr)ddnew=ddnew[1:thr,c(2,6,4)] else ddnew=ddnew[,c(2,6,4)]
    #ddnew=ddnew[1:5000,]
    ddnew=ddnew[1:min(thr,nrow(ddnew)),]
    write.table(ddnew,file=newff,quote=F,sep='\t',row.names=F)
  }
}

File: PrepareAdaptiveFile.py
#! usr/bin/python

import os
from os.path import exists
import numpy as np
import csv
from csv import reader
import sys

indir=sys.argv[1]
outdir=sys.argv[2]
thr=10000

#def PrepareAdaptiveFile(indir,outdir,thr=10000):
ffs=os.listdir(indir)
for ff in ffs:
    if('.tsv' not in ff):  
       continue
    ff0=ff
    if not os.path.exists(outdir):
       os.makedirs(outdir)   
    str1='TestReal-'   
    newff=outdir+'/'+str1+ff0
#    if exists(newff)==False:
#       continue
    csv_reader = reader(open(indir+'/'+ff,"r"), delimiter='\t', quotechar="\"") 
    ddnew=[] 
    for row in csv_reader:
      if '*' not in row[1]:
        if 'X' not in row[1]:
#         if '^C.+F$' not in row[1]:
          if (len(row[1])>=10) and (len(row[1])<=24):
            if 'unresolved' not in row[5]:
              if (row[1][0]=='C') and (row[1][-1]=='F'):
                  ddnew.append(row)
    ddnew=np.array(ddnew)      
    sorted_array = ddnew[ddnew[:,3].argsort()]   
    reverse_array = sorted_array[::-1]
    if len(reverse_array)>thr:
       col1=reverse_array[0:thr,1]
       col2=reverse_array[0:thr,5]
       col3=reverse_array[0:thr,3]
    else:
       col1=reverse_array[:,1]
       col2=reverse_array[:,5]
       col3=reverse_array[:,3]
    c=zip(col1,col2,col3)         
    with open(newff, 'w') as f:
       writer = csv.writer(f, delimiter='\t')
       writer.writerows(c)
